### ISSUE TITLE
GARD datamonkey no output

### ISSUE THREAD

**Issue by No-drama-llama** [veg/hyphy#1214]
Hi everybody,

I was running GARD on datamonkey using an amino acid sequence alignment. I get the notification email telling me my job is finished but when I click on the link I get nothing and it seems to be loading for a long time (link: http://datamonkey.org/gard/5f4e155b34c53e423ae7be08). GARD worked fine for me using codon alignments so I am not sure what the problem is. When I run the latest release of hyphy 2.5.18 on my computer it seems to run fine on the previously mentioned amino acid alignment. It is not finished yet (running for the past 2 days on 67 aa sequences) so I don't know if it will also give me an error.

Any advice would be welcome since I have a month to finish these analyses!

Best,
Anamarija

**Comment by stevenweaver** [comment]
Dear @No-drama-llama, 

Thank you for reporting this issue. There is a fix for not showing an error report at https://github.com/veg/datamonkey-js/issues/345 waiting for review before being pushed. 

In the meantime, this is the problem -


```
Analysis Description
--------------------
GARD : Genetic Algorithms for Recombination Detection. Implements a
heuristic approach to screening alignments of sequences for
recombination, by using the CHC genetic algorithm to search for
phylogenetic incongruence among different partitions of the data. The
number of partitions is determined using a step-up procedure, while the
placement of breakpoints is searched for with the GA. The best fitting
model (based on c-AIC) is returned; and additional post-hoc tests run to
distinguish topological incongruence from rate-variation. v0.2 adds and
spooling results to JSON after each breakpoint search conclusion

- __Requirements__: A sequence alignment.

- __Citation__: **Automated Phylogenetic Detection of Recombination Using a Genetic
Algorithm**, _Mol Biol Evol 23(10), 1891–1901

- __Written by__: Sergei L Kosakovsky Pond

- __Contact Information__: spond@temple.edu

- __Analysis Version__: 0.2

type: nucleotide
rv: GDD
>How many site rate classes to use (permissible range = [2,10], default value = 4, integer): rate-classes: 3
Error:

Master node received an error:The declared model alphabet '{
{"A", "C", "G", "T"}
}' does not match the gard.filter filter: '{
{"A", "C", "D", "E", "F", "G", "H", "I", "K", "L", "M", "N", "P", "Q", "R", "S", "T", "V", "W", "Y"}
}' in call to assert(Columns(alphabet)==Columns(givenAlphabet)&&model.MatchAlphabets(givenAlphabet,alphabet), "The declared model alphabet '"+alphabet+"' does not match the `filter` filter: '"+givenAlphabet+"'");

Function call stack
1 :  [namespace = VeKeNVoH] assert(Columns(alphabet)==Columns(givenAlphabet)&&model.MatchAlphabets(givenAlphabet,alphabet), "The declared model alphabet '"+alphabet+"' does not match the `filter` filter: '"+givenAlphabet+"'");

        Keyword arguments:
                {
                 "model":"JTT"
                }
-------
2 :  models.generic.AttachFilter(model.generic.DefineModel.model,data_filter);
-------
3 :  gard.model=model.generic.DefineModel(gard.model.generator,"gard.overallModel",{"0":"terms.global"},"gard.filter",null);
-------
```

Datamonkey is not supporting amino-acid at this time, despite it being stated as an option. We will address this issue immediately. 

Best,
Steven

Best,
Steven

**Comment by No-drama-llama** [comment]
Dear Steven,

Thank you for the prompt reply and the effort to fix this, much appreciated! When you do address the issue can you give me a heads up so I can give GARD another try?

Best,
Anamarija

**Comment by stevenweaver** [comment]
Dear @No-drama-llama, 

Absolutely, I'm bumping this up in priority now and will implement a fix ASAP. I will also bypass review and push given that 14 day window is excessive. 

Best,
Steven

**Comment by No-drama-llama** [comment]
Dear Steven,

Thank you so much! 

Best,
Anamarija

**Comment by No-drama-llama** [comment]
Hi,

First of all, thank you very much for implementing the GARD protein data analysis in datamonkey. Really appreciate it!

I was hoping you can help me with two analyses that are giving me problems. I can't send you the links because all the job pages are down with the 502 Bad Gateway error today, so I will explain myself. 

I am running one analysis on the datamonkey server-GARD, with 60 sequences and 341 sites (1448858.new-silverback) that has been running for the past 60 hours. It seems too long, i ran another analysis with the same number of sequences and 1578 sites that finished in a few hours. Is this normal or there might be an error?

The other analysis has 68 sequences and 3087 sites but it always gives me an error on the server. When I tried running it locally I get this error: 
"### Performing an exhaustive single breakpoint analysis
Error:
Invalid distance matrix passed to NeighborJoin.". I am not sure what the problem is since I specify "--type amino-acid" and it reads it correctly. I have used the same alignment for various other phylogenetic analyses and it hasn't been giving me any problems.

Any help is much appreciated and have a nice weekend!
Best,
Anamarija

**Comment by spond** [comment]
Dear @No-drama-llama,

Please take a look at the code in the `develop` branch of HyPhy. It fixes the issue your data has revealed, and will be included in the version 2.5.19 release.

Best,
Sergei
